### DISCUSSION TITLE
msr file parsing error

### DISCUSSION THREAD

**Discussion by wapayze** [GeoscienceAustralia/DynAdjust#210] [Q&A] [answered]
Hi everyone,

I'm attempting to add Epoch and ReferenceFrame information to historical GNSS measurements in a msr.xml file. After adding them in I get this error:
+ Parsing:
  ELLIOTTmsr.xml...    67%
- Error: instance document parsing failed
  - line 60, column 19, severity 1
  - no declaration found for element 'ReferenceFrame'

If I remove the ReferenceFrame and just try to bring in the Epoch I get the same error. 
  
I've compared the msr.xml file to others where Epoch and ReferenceFrame are being parsed and I can't see why it wouldn't be working. The file does work before my editing. Any ideas?

**Comment by BatchelorJ** [comment, marked as answer]
Hi @wapayze, are you able to send a copy of the ELLIOTTmsr.xml file with the Reference Frame and Epoch labels in? I can then have a crack at testing and see if I can help.

**Reply by wapayze** [reply]
Cheers. I have come up with a bit of workaround that seems to work but would still like find out why the error pops up. The workaround is to use -r on import to set the reference frame, use --export-xml to get new versions of the files, and then just find/replace all the epochs. The downside to this is that comments disappear in the exported xml files.

I've attached 3 files. orig_ELLIOT is the original file, edited_ELLIOT is the one I manually edited and bringing up the error on emport, and --export_ELLIOT is the workaround.

[ELLIOTmsr_files.zip](https://github.com/icsm-au/DynAdjust/files/10928529/ELLIOTmsr_files.zip)

**Reply by wapayze** [reply]
I should have added the station files. My bad.
[ELLIOTTstn.zip](https://github.com/icsm-au/DynAdjust/files/10928560/ELLIOTTstn.zip)

**Reply by BatchelorJ** [reply]
Hi @wapayze, thanks for sending those files through. I've done some testing and think the problem could be that you're using an old version of DynaML.xsd that doesn't have a declaration for Reference Frame or Epoch. Can you:
1. find the DynaML.xsd file you're using
2. see if at line 38 it says <xs:element ref="ReferenceFrame" minOccurs="0"/>
3. if it doesn't, replace with this version https://github.com/icsm-au/DynAdjust/blob/master/sampleData/DynaML.xsd

**Reply by wapayze** [reply]
Thanks, @BatchelorJ . That was the issue. We generally use the new one, I must have grabbed a historical one when playing with historical files.
